lib/smart_proxy_dynflow_core/callback.rb

```diff
@@ -1,23 +1,52 @@
 require 'rest-client'
 
+# rubocop:disable Lint/HandleExceptions
 begin
   require 'smart_proxy_dynflow/callback'
 rescue LoadError
 end
+# rubocop:enable Lint/HandleExceptions
 
 module SmartProxyDynflowCore
   module Callback
-    class Action < Dynflow::Action
-      def plan(callback, data)
-        plan_self(:callback => callback, :data => data)
-      end
+    class Request
+      class << self
+        def send_to_foreman_tasks(callback_info, data)
+          self.new.callback(self.prepare_payload(callback_info, data))
+        end
 
-      def run
-        Request.send_to_foreman_tasks(input[:callback], input[:data])
+        private
+
+        def prepare_payload(callback, data)
+          { :callback => callback, :data => data }.to_json
+        end
+
+        # rubocop:disable Metrics/PerceivedComplexity
+        def ssl_options
```

Comment:
Seems like `ssl_option` actually shouldn't be private, as it's being used here https://github.com/theforeman/smart_proxy_dynflow/pull/57/files#diff-3a71619fb07bb1f56160a122b3b42900R62 as `self.class.ssl_options`